src/rtlib/utf_convto_wchar.c

```diff
@@ -313,13 +333,6 @@ static FB_WCHAR *hUTF16ToWChar( const UTF_16 *src, FB_WCHAR *dst, ssize_t *chars
 		break;
 
 	case sizeof( UTF_16 ):
-		if( dst == NULL ) {
-			dst = malloc( (*chars + 1) * sizeof( UTF_16 ) );
-			memcpy( dst, src, *chars * sizeof( UTF_16 ) );
-			dst[*chars] = 0;
-		} else {
-			memcpy( dst, src, *chars * sizeof( UTF_16 ) );
-		}
 		break;
```

Comment:
missing call to `hUTF16ToUTF16()` here, `dst = hUTF16ToUTF16( src, dst, chars );` (found by compiler warning)